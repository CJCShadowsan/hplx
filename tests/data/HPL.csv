n,nb,p,q,time,gflops
1000,32,1,4,2.26,0.2959
1000,64,1,4,1.17,0.57175
1000,128,1,4,0.54,1.2289
1000,196,1,4,0.42,1.5851
1000,256,1,4,0.31,2.1411
5000,32,1,4,12.78,6.5248
5000,64,1,4,7.47,11.158
5000,128,1,4,4.51,18.485
5000,196,1,4,3.41,24.476
5000,256,1,4,3.63,22.979
10000,32,1,4,32.16,20.736
10000,64,1,4,25.21,26.451
10000,128,1,4,19.11,34.896
10000,196,1,4,17.39,38.353
10000,256,1,4,17.62,37.845
20000,32,1,4,153.17,34.825
20000,64,1,4,122.95,43.382
20000,128,1,4,119.6,44.6
20000,196,1,4,112.72,47.319
20000,256,1,4,114.56,46.562
1000,32,2,2,3.41,0.19567
1000,64,2,2,2.54,0.26287
1000,128,2,2,1.47,0.45523
1000,196,2,2,1.06,0.63037
1000,256,2,2,0.76,0.87372
5000,32,2,2,22.22,3.7522
5000,64,2,2,15.86,5.2558
5000,128,2,2,12.1,6.8889
5000,196,2,2,10.64,7.8337
5000,256,2,2,8.13,10.256
10000,32,2,2,55.63,11.988
10000,64,2,2,45.96,14.507
10000,128,2,2,33.46,19.926
10000,196,2,2,32.9,20.268
10000,256,2,2,29.38,22.694
20000,32,2,2,206.99,25.769
20000,64,2,2,179.72,29.679
20000,128,2,2,157.04,33.966
20000,196,2,2,138.27,38.577
20000,256,2,2,153.52,34.745
