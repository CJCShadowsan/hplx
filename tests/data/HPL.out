================================================================================
HPLinpack 2.3  --  High-Performance Linpack benchmark  --   December 2, 2018
Written by A. Petitet and R. Clint Whaley,  Innovative Computing Laboratory, UTK
Modified by Piotr Luszczek, Innovative Computing Laboratory, UTK
Modified by Julien Langou, University of Colorado Denver
================================================================================

An explanation of the input/output parameters follows:
T/V    : Wall time / encoded variant.
N      : The order of the coefficient matrix A.
NB     : The partitioning blocking factor.
P      : The number of process rows.
Q      : The number of process columns.
Time   : Time in seconds to solve the linear system.
Gflops : Rate of execution for solving the linear system.

The following parameter values will be used:

N      :    1000     5000    10000    20000 
NB     :      32       64      128      196      256 
PMAP   : Row-major process mapping
P      :       1        2 
Q      :       4        2 
PFACT  :   Right 
NBMIN  :       4 
NDIV   :       2 
RFACT  :   Crout 
BCAST  :  1ringM 
DEPTH  :       1 
SWAP   : Mix (threshold = 64)
L1     : transposed form
U      : transposed form
EQUIL  : yes
ALIGN  : 8 double precision words

--------------------------------------------------------------------------------

- The matrix A is randomly generated for each test.
- The following scaled residual check will be computed:
      ||Ax-b||_oo / ( eps * ( || x ||_oo * || A ||_oo + || b ||_oo ) * N )
- The relative machine precision (eps) is taken to be               1.110223e-16
- Computational tests pass if scaled residuals are less than                16.0

================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4        1000    32     1     4               2.26             2.9590e-01
HPL_pdgesv() start time Mon Dec  9 11:54:52 2024

HPL_pdgesv() end time   Mon Dec  9 11:54:54 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   3.71130207e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4        1000    64     1     4               1.17             5.7175e-01
HPL_pdgesv() start time Mon Dec  9 11:54:54 2024

HPL_pdgesv() end time   Mon Dec  9 11:54:55 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   4.64985389e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4        1000   128     1     4               0.54             1.2289e+00
HPL_pdgesv() start time Mon Dec  9 11:54:56 2024

HPL_pdgesv() end time   Mon Dec  9 11:54:56 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   6.30706826e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4        1000   196     1     4               0.42             1.5851e+00
HPL_pdgesv() start time Mon Dec  9 11:54:56 2024

HPL_pdgesv() end time   Mon Dec  9 11:54:57 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   5.01991147e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4        1000   256     1     4               0.31             2.1411e+00
HPL_pdgesv() start time Mon Dec  9 11:54:57 2024

HPL_pdgesv() end time   Mon Dec  9 11:54:57 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   6.43578393e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4        5000    32     1     4              12.78             6.5248e+00
HPL_pdgesv() start time Mon Dec  9 11:54:58 2024

HPL_pdgesv() end time   Mon Dec  9 11:55:11 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   1.60164959e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4        5000    64     1     4               7.47             1.1158e+01
HPL_pdgesv() start time Mon Dec  9 11:55:12 2024

HPL_pdgesv() end time   Mon Dec  9 11:55:19 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   1.67255559e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4        5000   128     1     4               4.51             1.8485e+01
HPL_pdgesv() start time Mon Dec  9 11:55:20 2024

HPL_pdgesv() end time   Mon Dec  9 11:55:25 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   2.42484764e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4        5000   196     1     4               3.41             2.4476e+01
HPL_pdgesv() start time Mon Dec  9 11:55:26 2024

HPL_pdgesv() end time   Mon Dec  9 11:55:29 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   2.60768875e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4        5000   256     1     4               3.63             2.2979e+01
HPL_pdgesv() start time Mon Dec  9 11:55:30 2024

HPL_pdgesv() end time   Mon Dec  9 11:55:34 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   2.79906612e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4       10000    32     1     4              32.16             2.0736e+01
HPL_pdgesv() start time Mon Dec  9 11:55:36 2024

HPL_pdgesv() end time   Mon Dec  9 11:56:08 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   1.50062383e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4       10000    64     1     4              25.21             2.6451e+01
HPL_pdgesv() start time Mon Dec  9 11:56:12 2024

HPL_pdgesv() end time   Mon Dec  9 11:56:37 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   1.83100072e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4       10000   128     1     4              19.11             3.4896e+01
HPL_pdgesv() start time Mon Dec  9 11:56:41 2024

HPL_pdgesv() end time   Mon Dec  9 11:57:00 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   1.62217584e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4       10000   196     1     4              17.39             3.8353e+01
HPL_pdgesv() start time Mon Dec  9 11:57:04 2024

HPL_pdgesv() end time   Mon Dec  9 11:57:21 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   1.90777366e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4       10000   256     1     4              17.62             3.7845e+01
HPL_pdgesv() start time Mon Dec  9 11:57:25 2024

HPL_pdgesv() end time   Mon Dec  9 11:57:42 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   1.82673898e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4       20000    32     1     4             153.17             3.4825e+01
HPL_pdgesv() start time Mon Dec  9 11:57:51 2024

HPL_pdgesv() end time   Mon Dec  9 12:00:24 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   8.73000938e-04 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4       20000    64     1     4             122.95             4.3382e+01
HPL_pdgesv() start time Mon Dec  9 12:00:38 2024

HPL_pdgesv() end time   Mon Dec  9 12:02:41 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   8.67653925e-04 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4       20000   128     1     4             119.60             4.4600e+01
HPL_pdgesv() start time Mon Dec  9 12:02:55 2024

HPL_pdgesv() end time   Mon Dec  9 12:04:55 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   9.00061765e-04 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4       20000   196     1     4             112.72             4.7319e+01
HPL_pdgesv() start time Mon Dec  9 12:05:09 2024

HPL_pdgesv() end time   Mon Dec  9 12:07:01 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   1.07281746e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4       20000   256     1     4             114.56             4.6562e+01
HPL_pdgesv() start time Mon Dec  9 12:07:16 2024

HPL_pdgesv() end time   Mon Dec  9 12:09:10 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   1.18290302e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4        1000    32     2     2               3.41             1.9567e-01
HPL_pdgesv() start time Mon Dec  9 12:09:18 2024

HPL_pdgesv() end time   Mon Dec  9 12:09:21 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   3.50817264e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4        1000    64     2     2               2.54             2.6287e-01
HPL_pdgesv() start time Mon Dec  9 12:09:21 2024

HPL_pdgesv() end time   Mon Dec  9 12:09:24 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   3.89364928e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4        1000   128     2     2               1.47             4.5523e-01
HPL_pdgesv() start time Mon Dec  9 12:09:24 2024

HPL_pdgesv() end time   Mon Dec  9 12:09:26 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   5.98527906e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4        1000   196     2     2               1.06             6.3037e-01
HPL_pdgesv() start time Mon Dec  9 12:09:26 2024

HPL_pdgesv() end time   Mon Dec  9 12:09:27 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   6.72539421e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4        1000   256     2     2               0.76             8.7372e-01
HPL_pdgesv() start time Mon Dec  9 12:09:27 2024

HPL_pdgesv() end time   Mon Dec  9 12:09:28 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   5.93164753e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4        5000    32     2     2              22.22             3.7522e+00
HPL_pdgesv() start time Mon Dec  9 12:09:28 2024

HPL_pdgesv() end time   Mon Dec  9 12:09:51 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   1.63014967e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4        5000    64     2     2              15.86             5.2558e+00
HPL_pdgesv() start time Mon Dec  9 12:09:52 2024

HPL_pdgesv() end time   Mon Dec  9 12:10:07 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   1.83281693e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4        5000   128     2     2              12.10             6.8889e+00
HPL_pdgesv() start time Mon Dec  9 12:10:08 2024

HPL_pdgesv() end time   Mon Dec  9 12:10:20 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   2.36123876e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4        5000   196     2     2              10.64             7.8337e+00
HPL_pdgesv() start time Mon Dec  9 12:10:22 2024

HPL_pdgesv() end time   Mon Dec  9 12:10:32 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   2.28991971e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4        5000   256     2     2               8.13             1.0256e+01
HPL_pdgesv() start time Mon Dec  9 12:10:33 2024

HPL_pdgesv() end time   Mon Dec  9 12:10:41 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   3.64030045e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4       10000    32     2     2              55.63             1.1988e+01
HPL_pdgesv() start time Mon Dec  9 12:10:44 2024

HPL_pdgesv() end time   Mon Dec  9 12:11:39 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   1.31020470e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4       10000    64     2     2              45.96             1.4507e+01
HPL_pdgesv() start time Mon Dec  9 12:11:43 2024

HPL_pdgesv() end time   Mon Dec  9 12:12:29 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   1.50754142e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4       10000   128     2     2              33.46             1.9926e+01
HPL_pdgesv() start time Mon Dec  9 12:12:32 2024

HPL_pdgesv() end time   Mon Dec  9 12:13:06 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   1.58215262e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4       10000   196     2     2              32.90             2.0268e+01
HPL_pdgesv() start time Mon Dec  9 12:13:09 2024

HPL_pdgesv() end time   Mon Dec  9 12:13:42 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   2.17014812e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4       10000   256     2     2              29.38             2.2694e+01
HPL_pdgesv() start time Mon Dec  9 12:13:46 2024

HPL_pdgesv() end time   Mon Dec  9 12:14:15 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   2.23043002e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4       20000    32     2     2             206.99             2.5769e+01
HPL_pdgesv() start time Mon Dec  9 12:14:24 2024

HPL_pdgesv() end time   Mon Dec  9 12:17:51 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   9.43455695e-04 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4       20000    64     2     2             179.72             2.9679e+01
HPL_pdgesv() start time Mon Dec  9 12:18:05 2024

HPL_pdgesv() end time   Mon Dec  9 12:21:05 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   8.70709361e-04 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4       20000   128     2     2             157.04             3.3966e+01
HPL_pdgesv() start time Mon Dec  9 12:21:19 2024

HPL_pdgesv() end time   Mon Dec  9 12:23:56 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   1.01134928e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4       20000   196     2     2             138.27             3.8577e+01
HPL_pdgesv() start time Mon Dec  9 12:24:10 2024

HPL_pdgesv() end time   Mon Dec  9 12:26:28 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   1.11208880e-03 ...... PASSED
================================================================================
T/V                N    NB     P     Q               Time                 Gflops
--------------------------------------------------------------------------------
WR11C2R4       20000   256     2     2             153.52             3.4745e+01
HPL_pdgesv() start time Mon Dec  9 12:26:42 2024

HPL_pdgesv() end time   Mon Dec  9 12:29:15 2024

--------------------------------------------------------------------------------
||Ax-b||_oo/(eps*(||A||_oo*||x||_oo+||b||_oo)*N)=   1.29974536e-03 ...... PASSED
================================================================================

Finished     40 tests with the following results:
             40 tests completed and passed residual checks,
              0 tests completed and failed residual checks,
              0 tests skipped because of illegal input values.
--------------------------------------------------------------------------------

End of Tests.
================================================================================
